internal/pyxis/pyxis_suite_test.go

```diff
@@ -44,129 +39,138 @@ func mustWrite(w io.Writer, s string) {
+			mustWrite(response, `{"_id":"updateImage","certified":true,"deleted":false,"image_id":"123456789abc"}`)
+		case request.Method == http.MethodPatch && request.Header["X-Api-Key"][0] == "my-update-image-failure-api-token":
+			response.WriteHeader(http.StatusInternalServerError)
+		case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-image-409-api-token":
+			response.WriteHeader(http.StatusConflict)
+		case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-bad-401-image-api-token":
+			response.WriteHeader(http.StatusConflict)
+		case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-bad-image-api-token":
+			response.WriteHeader(http.StatusConflict)
+		case request.Method == http.MethodGet && request.Header["X-Api-Key"][0] == "my-bad-401-image-api-token":
+			logger.V(log.TRC).Info("get with 401")
+			response.WriteHeader(http.StatusUnauthorized)
+		case request.Header["X-Api-Key"][0] == "my-bad-image-api-token":
+			response.WriteHeader(http.StatusUnauthorized)
+		case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-bad-500-image-api-token":
+			response.WriteHeader(http.StatusInternalServerError)
+		case request.Header["X-Api-Key"][0] == "my-index-docker-io-project-api-token":
+			mustWrite(response, `{"_id": "blah", "architecture": "amd64", "object_type": "containerImage", "repositories": [ {"published": false, "registry": "docker.io", "repository": "my/repo", "tags": [{"name": "docker_io_v3"}]}]}`)
+		case request.Method == http.MethodPost:
+			mustWrite(response, responseString)
+		default:
+			mustWrite(response, fmt.Sprintf(`{"data":[%s]}`, responseString))
+		}
 	}
 }
 
-func (p *pyxisRPMManifestHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
-	log.L().Trace("In the RPM Manifest ServeHTTP")
-	response.Header().Set("Content-Type", "application/json")
-	if request.Body != nil {
-		defer request.Body.Close()
-	}
-	switch {
-	case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-bad-rpmmanifest-409-api-token":
-		response.WriteHeader(http.StatusConflict)
-		mustWrite(response, `{"_id":"foo"}`)
-	case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-bad-rpmmanifest-401-api-token":
-		response.WriteHeader(http.StatusConflict)
-	case request.Method == http.MethodGet && request.Header["X-Api-Key"][0] == "my-bad-rpmmanifest-401-api-token":
-		response.WriteHeader(http.StatusUnauthorized)
-	case request.Header["X-Api-Key"][0] == "my-bad-rpmmanifest-api-token":
-		response.WriteHeader(http.StatusUnauthorized)
-	case request.Method == http.MethodPost && request.Header["X-Api-Key"][0] == "my-update-image-success-api-token":
-		mustWrite(response, `{"_id":"updateImage"}`)
-	default:
-		mustWrite(response, `{"_id":"blah"}`)
+func pyxisRPMManifestHandler(ctx context.Context) func(http.ResponseWriter, *http.Request) {
```

Comment:
NIT: could the return value have been http.HandlerFunc?

Reply:
Yes. It could.